instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/InstrumenterBuilder.java

```diff
@@ -189,6 +235,34 @@
     return constructor.create(this);
   }
 
+  SpanSuppressionStrategy getSpanSuppressionStrategy() {
+    if (!enableSpanSuppressionByType) {
+      // if not enabled, preserve current behavior, not distinguishing types
+      return SpanSuppressionStrategy.SUPPRESS_ALL_NESTED_OUTGOING_STRATEGY;
+    }
+
+    List<SpanKey> spanKeys = spanKeysFromAttributeExtractor(this.attributesExtractors);
+    return SpanSuppressionStrategy.from(spanKeys);
+  }
+
+  private static List<SpanKey> spanKeysFromAttributeExtractor(
+      List<? extends AttributesExtractor<?, ?>> attributesExtractors) {
+
+    List<SpanKey> spanKeys = new ArrayList<>();
+    for (AttributesExtractor<?, ?> attributeExtractor : attributesExtractors) {
+      if (attributeExtractor instanceof HttpAttributesExtractor) {
```

Comment:
Does this order matter? If so, let's add a comment to make sure that's not forgotten. If not, consider using `Set` instead of `List` to codify that fact - in fact while we don't expect it in practice, we may have multiple extractors of the same type registered so that would also get deduped.